torch/optim/adadelta.py

```diff
@@ -200,6 +219,11 @@ def adadelta(
     if foreach is None:
         _, foreach = _default_to_fused_or_foreach(params, differentiable, use_fused=False)
 
+    # If compiling, the compiler will handle cudagraph checks, see note [torch.compile x capturable]
+    if not torch._utils.is_compiling() and capturable:
```

Comment:
Let's move this check to later in the individual functions (single_tensor and multi_tensor).